lib/gollum/app.rb

```diff
@@ -241,22 +241,9 @@ class App < Sinatra::Base
           tempfile = params[:file][:tempfile]
         end
         halt 500 unless tempfile.is_a? Tempfile
+        
+        dir = wiki.per_page_uploads ? find_per_page_upload_subdir(request.referer, request.host_with_port, wiki.base_path) : 'uploads'
 
-        if wiki.per_page_uploads
-          dir = request.referer.match(/^https?:\/\/#{request.host_with_port}\/(.*)/)[1]          
-          # remove base path if it is set
-          dir.sub!(/^#{wiki.base_path}/, '') if wiki.base_path
```

Comment:
The regex for this `sub!` would not match because 1. `base_path` usually starts with a slash and 2. on https://github.com/gollum/gollum/pull/1831/files#diff-65eee83dd06f55c49a91cd71e716ae5e5ef09b353bf7472d4079ffdc175c395aL246 we were selecting the part of the referer after that slash